lib/services_cli.rb

```diff
@@ -396,5 +289,94 @@ def kill(service)
       end
       ohai "Successfully stopped `#{service.name}` via #{service.label}"
     end
+
+    def install_service_file(service)
+      temp = Tempfile.new(service.label)
+      temp << service.generate_plist(@plist)
+      temp.flush
+
+      rm service.dest if service.dest.exist?
+      service.dest_dir.mkpath unless service.dest_dir.directory?
+      cp temp.path, service.dest
+
+      # Clear tempfile.
+      temp.close
+
+      chmod 0644, service.dest
+    end
+
+    # protections to avoid users editing root services
+    def take_root_ownership(service)
+      return unless root?
+
+      chown "root", "admin", service.dest
+      plist_data = service.dest.read
+      plist = begin
+        Plist.parse_xml(plist_data)
+      rescue
+        nil
+      end
+      return unless plist
+
+      root_paths = []
+
+      program_location = plist["ProgramArguments"]&.first
+      key = "first ProgramArguments value"
+      if program_location.blank?
+        program_location = plist["Program"]
+        key = "Program"
+      end
+
+      if program_location.present?
+        Dir.chdir("/") do
+          if File.exist?(program_location)
+            program_location_path = Pathname(program_location).realpath
+            root_paths += [
+              program_location_path,
+              program_location_path.parent.realpath,
+            ]
+          else
+            opoo <<~EOS
+              #{service.name}: the #{key} does not exist:
+                #{program_location}
+            EOS
+          end
+        end
+      end
+      if (formula = service.formula)
+        root_paths += [
+          formula.opt_prefix,
+          formula.linked_keg,
+          formula.bin,
+          formula.sbin,
+        ]
+      end
+      root_paths = root_paths.sort.uniq.select(&:exist?)
+
+      opoo <<~EOS
+        Taking root:admin ownership of some #{service.formula} paths:
+          #{root_paths.join("\n  ")}
+        This will require manual removal of these paths using `sudo rm` on
+        brew upgrade/reinstall/uninstall.
+      EOS
+      chown "root", "admin", root_paths
+      chmod "+t", root_paths
+    end
+
+    def launchctl_load(service, enable)
```

Comment:
```suggestion
    def launchctl_load(service, enable:)
```

makes the callers a bit more readable